### PR TITLE
Machine mount - add mount list handler to machine group

### PR DESCRIPTION
**Pull request by ppknap**
Depends on: ~#10377~

**Code coverage**:  51.3% of statements

## How Has This Been Tested?
Unit tests.

## Screenshots (if appropriate):
none

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: go/src/koding/klient/machine/machinegroup/mount.go [hunk 1]
```diff
@@ -141,3 +142,118 @@ func (g *Group) AddMount(req *AddMountRequest) (res *AddMountResponse, err error
 		MountID: mountID,
 	}, nil
 }
+
+// ListMountRequest defines machine group mount list request.
+type ListMountRequest struct {
+	// ID is an optional identifier for the remote machine. If set, only
+	// mounts related to this machine will be returned.
+	ID machine.ID `json:"id"`
```
Comment (rjeczalik) [inline go/src/koding/klient/machine/machinegroup/mount.go]: ,omitempty

File: go/src/koding/klient/machine/machinegroup/mount.go [hunk 2]
```diff
@@ -141,3 +142,118 @@ func (g *Group) AddMount(req *AddMountRequest) (res *AddMountResponse, err error
...
+
+	// MountID is an optional identifier of a mount which is meant to be listed.
+	MountID mount.ID `json:"mountID"`
```
Comment (rjeczalik) [inline go/src/koding/klient/machine/machinegroup/mount.go]: ditto

File: go/src/koding/klient/machine/machinegroup/mount.go [hunk 3]
```diff
@@ -141,3 +142,118 @@ func (g *Group) AddMount(req *AddMountRequest) (res *AddMountResponse, err error
...
+		m  mount.Mount
+		id machine.ID
+	}
+
+	// Get list of requested mounts with machine ID filter.
+	var mms = make(map[mount.ID]mountsMachine)
+	if req.ID != "" {
+		mounts, err := g.mount.All(req.ID)
+		if err != nil {
+			return res, nil
```
Comment (rjeczalik) [inline go/src/koding/klient/machine/machinegroup/mount.go]: Why the early return here? Shouldn't we try req.MountID as well? If req.ID and req.MountID are exclusive, then some check is missing.

**Review by rjeczalik — APPROVED** [review]

File: go/src/koding/klient/machine/machinegroup/mount.go [hunk 4]
```diff
@@ -141,3 +142,118 @@ func (g *Group) AddMount(req *AddMountRequest) (res *AddMountResponse, err error
...
+		}
+
+		m := mount.Mount{}
+		if mounts, err := g.mount.All(req.ID); err == nil {
+			m = mounts[req.MountID]
+		}
+
+		mms = map[mount.ID]mountsMachine{
+			req.MountID: mountsMachine{m: m, id: id},
+		}
```
Comment (rjeczalik) [inline go/src/koding/klient/machine/machinegroup/mount.go]: If `g.mount.All` failed we'd be adding an empty mount here.

File: go/src/koding/klient/machine/machinegroup/mount.go (hunk 3 above)
Comment (rjeczalik) [inline go/src/koding/klient/machine/machinegroup/mount.go]: If an empty mount from L205 is eliminated then indeed we could early return here.

File: go/src/koding/klient/machine/machinegroup/mount.go [hunk 5]
```diff
@@ -141,3 +142,118 @@ func (g *Group) AddMount(req *AddMountRequest) (res *AddMountResponse, err error
...
+type ListMountResponse struct {
+	// Mounts is a map that contains machine aliases as keys and created mount
+	// infos which store the current synchronization status of mount.
+	Mounts map[string][]sync.Info `json:"mounts"`
+}
+
+// ListMount checks the status of mounts and returns their infos. This function
+// guarantees that if returned error is nil, response Mounts field is always
+// non-nil. It doesn't fail if provided filters contain incorrect values.
+func (g *Group) ListMount(req *ListMountRequest) (*ListMountResponse, error) {
```
Comment (cihangir) [inline go/src/koding/klient/machine/machinegroup/mount.go]: could you separate this function into smaller logical parts?, it is hard ( for me ) to follow it

File: go/src/koding/klient/machine/machinegroup/mount_test.go [hunk 6]
```diff
@@ -126,3 +129,181 @@ func TestAddMount(t *testing.T) {
...
+				ID:      idB,
+				MountID: "unknown",
+			},
+			ConcatIDsInfo: map[string]sync.Info{},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			listMountRes, err := g.ListMount(&test.LMReq)
```
Comment (cihangir) [inline go/src/koding/klient/machine/machinegroup/mount_test.go]: could you add t.Parallel here?

**Comment by ppknap** [comment]
@cihangir I'm merging this for now in order to rebase #10383. If you have more change requests, write them here and I will apply them in next PR.

**Pull request merged**